From 702145d7e8457102023194d67d3abcb6ce2331b4 Mon Sep 17 00:00:00 2001
From: "Ryuhoeun.ryu@daum.net" <Ryuhoeun.ryu@daum.net>
Date: Sat, 11 Aug 2018 22:48:27 +0900
Subject: [PATCH] events: remove snd_soc_codec above kernel version 4.17.0

The struct snd_soc_codec removes from linux kernel. Please see the
following commit in linux kernel.
"ASoC: trace: remove snd_soc_codec" (8ba05d770dc46df76900d4e17c312d62c83bed6f)

Signed-off-by: Hoeun Ryu <Hoeun Ryu<hoeun.ryu@daum.net>>
---
 instrumentation/events/lttng-module/asoc.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/instrumentation/events/lttng-module/asoc.h b/instrumentation/events/lttng-module/asoc.h
index 25136ec..f0aef25 100644
--- a/instrumentation/events/lttng-module/asoc.h
+++ b/instrumentation/events/lttng-module/asoc.h
@@ -13,7 +13,9 @@
 #ifndef _TRACE_ASOC_DEF
 #define _TRACE_ASOC_DEF
 struct snd_soc_jack;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,17,0))
 struct snd_soc_codec;
+#endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,1,0))
 struct snd_soc_platform;
 #endif
@@ -30,6 +32,7 @@ struct snd_soc_dapm_widget;
 #define CODEC_ID_FIELD id
 #endif
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,17,0))
 /*
  * Log register events
  */
@@ -108,6 +111,7 @@ LTTNG_TRACEPOINT_EVENT_INSTANCE_MAP(asoc_snd_soc_preg, snd_soc_preg_read,
 
 )
 #endif
+#endif /* (LINUX_VERSION_CODE < KERNEL_VERSION(4,17,0)) */
 
 LTTNG_TRACEPOINT_EVENT_CLASS(asoc_snd_soc_card,
 
@@ -326,6 +330,7 @@ LTTNG_TRACEPOINT_EVENT_MAP(snd_soc_jack_notify,
 	)
 )
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,17,0))
 LTTNG_TRACEPOINT_EVENT_MAP(snd_soc_cache_sync,
 
 	asoc_snd_soc_cache_sync,
@@ -342,6 +347,7 @@ LTTNG_TRACEPOINT_EVENT_MAP(snd_soc_cache_sync,
 		ctf_integer(int, id, codec->CODEC_ID_FIELD)
 	)
 )
+#endif
 
 #endif /* LTTNG_TRACE_ASOC_H */
 
